src/test/java/bio/terra/janitor/app/controller/JanitorApiControllerTest.java

```diff
@@ -219,4 +218,204 @@ public void getResources() throws Exception {
             .collect(Collectors.toList()),
         Matchers.containsInAnyOrder(id1, id2));
   }
+
+  @Test
+  public void abandonThenBumpResources() throws Exception {
```

Comment:
[Side note / comment] Not directly related to this PR, but I feel like it's not quite right to be testing the service business logic from the controller level. It adds a fair amount of boilerplate, since we're dealing with things like serializing requests and tracking URL paths. I'm curious if there's been much discussion within or across teams about the aesthetic of what layer to run Spring-based unit tests. (We've had many such discussions on the AoU team, and I'm happy to share our experiences.)

Reply:
We haven't discussed the level of test. The 1st test we added is createResource test. 
At that time, I am not confident about the open api schema, so I want the test starts from controller then I can verify the url, query, response code all together.  
Seems WM doesn't have controller test but they have service test.
While Data Repo has controller test: https://github.com/DataBiosphere/jade-data-repo/tree/develop/src/test/java/bio/terra/app/controller and that is where I learnt the test structure.
Happy to chat more about the test structure. Thanks!